### PR TITLE
Ziemleszcz/ipsec

### PR DESCRIPTION
**Pull request by ziemleszcz**
This commit prepares IPSEC build, adds some extra info to /dev/ifstatus and creates new file /dev/ipsecdev to monitor ethernet cable status.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imx6ull).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

Requires (to work, not to build):
https://gerrit.phoesys.com/c/dataro/+/2864/1
https://gerrit.phoesys.com/c/dataro/+/2865/1

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: port/main.c [hunk 1]
```diff
@@ -121,10 +133,69 @@ static int writeStatus(char *buffer, size_t bufSize, size_t offset)
 			SNPRINTF_APPEND("%2s%d_media=%s\n", netif->name, netif->num, drv->media(netif));
 	}
 
+#if LWIP_DNS
+	for (i = 0; i < LWIP_DHCP_MAX_DNS_SERVERS; ++i)
+		SNPRINTF_APPEND("dns_%u=%s\n", i, inet_ntoa(*dns_getserver(i)));
```
Comment (nalajcie) [inline port/main.c:138]: not connected to this PR: what will happen if we would setup 2 ethernet ports to DHCP? I think everything will work apart from (potentially) DNS servers?

File: port/main.c [hunk 2]
```diff
@@ -49,6 +50,13 @@
 	} while (0)
 
 
+#if LWIP_IPSEC
+static int ipsecdev_disconnected;
```
Comment (nalajcie) [inline port/main.c:54]: I believe our coding convention is to put all global variables to one "common" struct.

I like the simplicity of this solution, but the naming is misleading (link monitoring is not connected to IPSec in any way, just the interface name is taken from `LWIP_IPSEC_DEV`).

I think we could either:
 - provide separate define for monitoring device (so the functionality would not be under `LWIP_IPSEC`)
 - add support for writing device name in runtime (so to start monitoring the device I would do `echo -n "en1" > /dev/linkmonitor`) and than start reading the file?

**Review by nalajcie — CHANGES_REQUESTED** [review]
I think we should change the naming (drop link monitoring and IPSec dependency). Other than that - I like the simplicity of the approach, regardless of the need to poll for the changes.

We could think of adding blocking read here (storing readers on wait queue and releasing them on link status change) in the future - but not right now as blocking reads can't be interrupted/canceled on our OS yet.

File: port/main.c (hunk 1 above)
Comment (ziemleszcz) [reply port/main.c:138]: DNS servers are overwritten by DHCP .

Comment (nalajcie) [reply port/main.c:140]: thx for the info

**Review by nalajcie — APPROVED** [review]
LGTM.

**Pull request merged**